Backend/src/KCTest.Infrastructure/Repositories/Repository.cs

```diff
@@ -17,8 +18,18 @@ public class Repository<TEntity> : IRepository<TEntity> where TEntity : BaseEnti
 
         public async Task AddAsync(TEntity entity) => await _entities.AddAsync(entity);
         public async Task UpdateAsync(TEntity entity) => await Task.Run(() => { _entities.Update(entity); });
-        public async Task<TEntity> GetByIdAsync(int id, IEnumerable<string> entitiesToInclude) => await _entities.Include(entitiesToInclude).FirstOrDefaultAsync(e => e.Id == id);
-        public async Task<IEnumerable<TEntity>> GetAllAsync(IEnumerable<Expression<Func<TEntity, bool>>> predicates, IEnumerable<string> entitiesToInclude) => await _entities.Filter(predicates).Include(entitiesToInclude).ToListAsync();
+        public async Task<TEntity> GetByIdAsync(int id, IEnumerable<string> entitiesToInclude) => await _entities.Include(entitiesToInclude)
+                                                                                                                 .FirstOrDefaultAsync(e => e.Id == id);
+        public async Task<IEnumerable<TEntity>> GetAllAsync(IEnumerable<Expression<Func<TEntity, bool>>> predicates,
+                                                            IEnumerable<string> entitiesToInclude) => await _entities.Filter(predicates)
+                                                                                                                     .Include(entitiesToInclude)
+                                                                                                                     .ToListAsync();
+        public async Task<IEnumerable<TEntity>> GetAllAsync(int skip, int limit, IEnumerable<Expression<Func<TEntity, bool>>> predicates,
+                                                                                 IEnumerable<string> entitiesToInclude) => await _entities.Skip(skip)
+                                                                                                                                          .Take(limit)
+                                                                                                                                          .Filter(predicates)
```

Comment:
You're paginating before applying the filter here. Pagination should be done after Filtering.